extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/HttpServerInitializer.java

```diff
@@ -0,0 +1,160 @@
+package io.quarkus.vertx.web.runtime;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CountDownLatch;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.Timing;
+import io.quarkus.vertx.web.Route;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+@Singleton
+public class HttpServerInitializer {
+
+    private static final Logger LOGGER = Logger.getLogger(HttpServerInitializer.class.getName());
+
+    private volatile Router router;
+
+    private volatile HttpServer httpServer;
+
+    @Inject
+    Vertx vertx;
+
+    @Inject
+    Event<Router> routerEvent;
+
+    void initialize(VertxHttpConfiguration vertxHttpConfiguration, Map<String, List<Route>> routeHandlers,
+            LaunchMode launchMode) {
+        Router router = Router.router(vertx);
+        router.route().handler(BodyHandler.create());
```

Comment:
In dev mode, we should launch the scan.

Reply:
I don't understand ;-). What "scan"?

Reply:
I was referring to the hot reload. It should be possible to have a route triggering it.

Reply:
I see. Yes, that would make sense.